equinox/nn/_embedding.py

```diff
@@ -89,3 +96,313 @@ def __call__(
                 "`eqx.nn.Embedding()(x)` should be called with a scalar index `x`. "
                 "Use `jax.vmap` if you would like to index with multiple values."
             )
+
+
+class RotaryPositionalEmbedding(Module, strict=True):
+    """
+    A rotary positional encoding module, as described in the paper
+    "RoFormer: Enhanced Transformer with Rotary Position Embedding". While
+    this module can be used in any context, it is particularly useful for
+    providing positional information to transformer models.
+
+    !!! example
+        The following example demonstrates how to use `RotaryPositionalEmbedding` in
+        a simple transformer model.
+        ```python
+
+        class TransformerBlock(eqx.nn.StatefulLayer):
```

Comment:
This doesn't need to be a stateful layer.
(That's used just as a marker for `eqx.nn.Sequential` to call the layer as `layer(x, state)` rather than `layer(x)`.)